src/io/hdf_archive.h

```diff
@@ -61,17 +63,17 @@ struct hdf_archive
    * @param c communicator
    * @param use_collective turn on/off collective
```

Comment:
Update the parameter name in the documentation description.  Something like "request_pio  Request parallel I/O" (so that the short 'pio' gets defined somewhere).
Also it might be good to describe what happens if parallel I/O is not available (and under what conditions parallel I/O is or is not available).